Orthtree/include/CGAL/Orthtree_traits_face_graph.h

```diff
@@ -0,0 +1,152 @@
+// Copyright (c) 2023  INRIA (France).
+// All rights reserved.
+//
+// This file is part of CGAL (www.cgal.org).
+//
+// $URL$
+// $Id$
+// SPDX-License-Identifier: GPL-3.0-or-later OR LicenseRef-Commercial
+//
+// Author(s)     : Sebastien LORIOT
+
+
+#ifndef CGAL_ORTHREE_TRAITS_FACE_GRAPH_H
+#define CGAL_ORTHREE_TRAITS_FACE_GRAPH_H
+
+#include <CGAL/license/Orthtree.h>
+
+#include <CGAL/Orthtree_traits_base_for_dimension.h>
+
+#include <boost/graph/graph_traits.hpp>
+#include <CGAL/Dimension.h>
+#include <CGAL/Bbox_3.h>
+
+namespace CGAL {
+
+/*!
+  \ingroup PkgOrthtreeTraits
+
+  The class `Orthtree_traits_face_graph` can be used as a template parameter of
+  the `Orthtree` class.
+
+  \tparam PolygonMesh a model of `FaceGraph`.
+  \tparam VertexPointMap a property map associating points to the vertices of `PolygonMesh`.
+
+  \cgalModels{OrthtreeTraits}
+  \sa `CGAL::Orthtree_traits_base_for_dimension<GeomTraits, DimensionTag>`
+*/
+template <class PolygonMesh, class VertexPointMap>
+struct Orthtree_traits_face_graph : public Orthtree_traits_base_for_dimension<
+  typename Kernel_traits<typename boost::property_traits<VertexPointMap>::value_type>::type,
+  Dimension_tag<3> > {
+
+  Orthtree_traits_face_graph(const PolygonMesh& pm, VertexPointMap vpm)
+    : m_pm(pm), m_vpm(vpm) {}
+
+  /// \name Types
+  /// @{
+
+  using Self = Orthtree_traits_face_graph<PolygonMesh, VertexPointMap>;
+  using Tree = Orthtree<Self>;
+
+  using Point_d = typename Self::Point_d;
+  using Dimension = typename Self::Dimension;
+  using Bbox_d = typename Self::Bbox_d;
+  using FT = typename Self::FT;
+  using Cartesian_const_iterator_d = typename Self::Cartesian_const_iterator_d;
+
+  using Node_data = std::vector<typename boost::graph_traits<PolygonMesh>::face_descriptor>;
+
+  using Geom_traits = typename Kernel_traits<Point_d>::type;
+
+  /// @}
+
+  /// \name Operations
+  /// @{
+
+  auto construct_root_node_bbox_object() const {
+    return [&]() -> Bbox_d {
+
+      std::array<FT, Dimension::value> min = {0.0, 0}, max = {0.0, 0};
+      if (faces(m_pm).begin() != faces(m_pm).end()) {
+        const Point_d& p = get(m_vpm, *vertices(m_pm).begin());
+        min = {p.x(), p.y(), p.z()};
+        max = min;
+        for (auto v: vertices(m_pm)) {
+          const Point_d& p_v = get(m_vpm, v);
+          for (int i = 0; i < 3; ++i) {
+            if (p_v[i] < min[i]) min[i] = p_v[i];
+            if (p_v[i] > max[i]) max[i] = p_v[i];
+          }
+        }
+      }
+
+      return {std::apply(Self::construct_point_d_object(), min),
+              std::apply(Self::construct_point_d_object(), max)};
+    };
+  }
+
+  auto construct_root_node_contents_object() {
+    return [&]() -> Node_data {
+      return {faces(m_pm).begin(), faces(m_pm).end()};
+    };
+  }
+
+  auto distribute_node_contents_object() {
```

Comment:
This seems to be written for ``TriangleMesh` and not `PolygonMesh`.  What do we want?

Reply:
For now the implementation is restricted to triangles. As there is no use case so far, extending to polygons does not seem useful now.